### ISSUE TITLE
Problems with Scandinavian keyboard layout

### ISSUE THREAD

**Issue by eberhardweber** [midwan/amiberry#585]
I'm running Amiberry 3.1.0 on Retropie. 

I'm using a Scandinavian-layout (extra ÖÄÅ etc.) USB Logitech keyboard with my Raspberry Pi. I've used Raspi-Config to set the most approximate keyboard model and set the Finnish keyboard layout to be used in RetroPie. The keyboard layout does work perfectly fine in terminal usage and in other use cases.

In Amiberry, however, games that require keyboard input (like inputting passwords, in the game "Flood" for instance) none of the scandi-specific keys are recognized, and, as my biggest current problem, the "Enter" key instead inputs the letter "P" in text parsers.

In other words, I'm unable to input "Enter" commands at all. It does make some games unplayable. I tried connecting my Bluetooth keyboard as well, but since it's also a Scandi keyboard the result was the same. 

This sounds quite a bit like issue #558 but the thread didn't really help me get anywhere. I'm unfortunately not familiar with the Amiberry config files (what goes where) and I wasn't able to find what I was looking for in this respect in the Wiki (what cfg does what) so I'd be interested in knowing how I would go about binding the "Enter" (return?) key to "Enter" somehow to get going? :)

**Comment by midwan** [comment]
@eberhardweber 
Are you using Keyboard as Joystick on the second port?
In that case, some keys are mapped to "special" keys to help with games that use them - P would be one that is used for Pausing a game for example, so please check for that first.

**Comment by eberhardweber** [comment]
@midwan 
With port 2, I assume you mean the virtual "Parallel 1" port in the Amiberry settings? I have yet to configure the keyboard in Amiberry at all, the ports have been used for other devices (mouse and three keypads) so far. 

So far I've tried the following:
- Switch the mouse and keyboard USB slots around on the Raspberry (It's a 3B+)
- Change the keyboard locale from Finnish to a UK layout in Raspi-Config
- Clear the Parallel 1 (port 2) device (was a X360 keypad) in the Amiberry settings and set it to "None"

So far still P. I think you have the right idea here, sounds very plausible. Anything else you think I could try?

**Comment by midwan** [comment]
@eberhardweber 
No, I meant the 2nd port in the list (port 0 is the mouse, port 1 is the first joystick, and we have 2 more ports connected in the Parallel port). If you have "Keyboard as Joystick" there, try setting it to None, or to a real game controller (if you have one connected), to see if it changes the behavior you mentioned.

**Comment by eberhardweber** [comment]
@midwan 
None of the ports have been set to "Keyboard as Joystick" at any point in time while I've used Amiberry. All of them are occupied by other devices :)

**Comment by solskogen** [comment]
With a norwegian layout, I can press enter without getting 'P'

**Comment by eberhardweber** [comment]
@solskogen 
I'm pretty sure my problem isn't about the layout even though it seems like it. As I mentioned, I tried the UK layout and the problem persists with it. @midwan seems to have a good clue, but I'm not sure how to proceed troubleshooting the issue.

**Comment by midwan** [comment]
@eberhardweber 
This issue has been stale for quite a while now. Did you get anywhere further with it?
I've tried scandinavian keyboards and they seem to work normally here.

If you're still having this problem, please provide a screenshot of the current settings you have in the Input panel of Amiberry, while having the problem. If it's what I think it is, it will be visible there.

**Comment by midwan** [comment]
Closing due to inactivity, feel free to come back with more information if this is still an issue. :)
